### PR TITLE
fix: records update bugs and inefficiency

### PR DESCRIPTION
**Pull request by TBonnin**
## Describe your changes

Found a few small issues in the records update function as well as some inefficiency.

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/records/lib/models/records.ts [hunk 1]
```diff
@@ -322,24 +322,26 @@ export async function update({
             for (let i = 0; i < recordsWithoutDuplicates.length; i += BATCH_SIZE) {
                 const chunk = recordsWithoutDuplicates.slice(i, i + BATCH_SIZE);
 
-                updatedKeys.push(...(await getUpdatedKeys({ records: chunk, connectionId, model, trx })));
+                const chunkUpdatedKeys = await getUpdatedKeys({ records: chunk, connectionId, model, trx });
+                if (chunkUpdatedKeys.length === 0) {
+                    logger.warn(`Did not find records to update: ${{ connectionId, model }}`);
```
Comment (bodinsamuel) [inline packages/records/lib/models/records.ts]: logger.warn does not exists (types are wrong), it's logger.warning

File: packages/records/lib/models/records.ts [hunk 2]
```diff
@@ -322,24 +313,27 @@ export async function update({
             for (let i = 0; i < recordsWithoutDuplicates.length; i += BATCH_SIZE) {
                 const chunk = recordsWithoutDuplicates.slice(i, i + BATCH_SIZE);
 
-                updatedKeys.push(...(await getUpdatedKeys({ records: chunk, connectionId, model, trx })));
+                const oldRecords = await getRecordsToUpdate({ records: chunk, connectionId, model, trx });
```
Comment (TBonnin) [inline packages/records/lib/models/records.ts:316]: we used to make 2 queries to get the existing records to update. :(

File: packages/records/lib/models/records.ts [hunk 3]
```diff
@@ -322,24 +313,27 @@ export async function update({
...
                     }
 
-                    const { record: oldRecord, ...oldRecordRest } = rawOldRecord;
-
-                    const record = records.find((record) => record.external_id === rawOldRecord.external_id);
+                    const { json: newRecordData, ...newRecordRest } = inputRecord;
 
                     const newRecord: FormattedRecord = {
-                        ...oldRecordRest,
+                        ...newRecordRest,
```
Comment (TBonnin) [inline packages/records/lib/models/records.ts:333]: that was a bug. The only attributes that is different is the sync_job_id but was not updated

File: packages/records/lib/utils/encryption.ts [hunk 4]
```diff
@@ -14,7 +14,7 @@ function isEncrypted(data: UnencryptedRecordData | EncryptedRecordData): data is
     return 'encryptedValue' in data;
 }
 
-export function decryptRecord(record: FormattedRecord): UnencryptedRecordData {
+export function decryptRecordData(record: FormattedRecord): UnencryptedRecordData {
```
Comment (TBonnin) [inline packages/records/lib/utils/encryption.ts:17]: renaming to make it clearer that the function only return the unencrypted data not the entire record

File: packages/webhooks/lib/sync.ts [hunk 5]
```diff
@@ -69,7 +69,7 @@ export const sendSync = async ({
             responseResults?.added === 0 && responseResults?.updated === 0 && (responseResults.deleted === 0 || responseResults.deleted === undefined);
 
         if (!webhookSettings.on_sync_completion_always && noChanges) {
-            await logCtx?.info('There were no added, updated, or deleted results. No webhook sent, as per your environment settings');
+            await logCtx?.info(`There were no added, updated, or deleted results for model ${model}. No webhook sent, as per your environment settings`);
```
Comment (TBonnin) [inline packages/webhooks/lib/sync.ts:72]: unrelated but it is confusing in the logs to see multiple times the same message when sync is multi model

**Review by bodinsamuel — APPROVED** [review]

File: packages/records/lib/models/records.ts (hunk 1 above)
Comment (TBonnin) [reply packages/records/lib/models/records.ts]: old man yells at cloud

Comment (TBonnin) [reply packages/records/lib/models/records.ts]: old man yells at cloud (1)

**Pull request merged**